pkg/types/list.go

```diff
@@ -0,0 +1,39 @@
+package types
+
+type Set []string
```

Comment:
Using a `map` also has these two downsides:

* We need a constructor, which currently isn't required (which I like).
* Since our lovely golang map gets shuffled all the time, we get non-deterministic results (ie resources get nuked in a different order every time).

Reply:
With the current implementation it would remove both.

I am not eager to implement something just to match a name. We can rename it, if you suggest a better name.

Reply:
Oh yeah, you are totally right, it would remove both. Sorry about that.

Hmm..., naming things is hard ... how about `LazySet`. What do you think? It would imply it is not quite a super strict set but it certainly is a set. Feel free to stick with `Set` if you prefer.

Reply:
"lazy" always makes me think of lazy loading. What about `Collection` to remove any implication from the name?

Reply:
Good idea! I like it.

Reply:
renamed